platform/src/main/java/org/hillview/sketches/RandomSampling.java

```diff
@@ -0,0 +1,50 @@
+package org.hillview.sketches;
+
+import org.hillview.table.SmallTable;
+import org.hillview.table.rows.RowSnapshot;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class RandomSampling implements Serializable {
```

Comment:
But totalRows is unknown for this sketch, right? It requires a sketch in itself to count the total number of rows in the dataset.

I thought that if you give the sketch a sampling rate, you can't control the exact number of samples that you get. (Or at least there'll be rounding problems, I expect it might be off by +/- 1.) The advantage of this method (if it's correct) is that you don't need to know the total number of rows in the table, and that you don't get these rounding problems if you want to control the number of samples exactly.

Reply:
The table knows the totalRows.
Also, sample a big higher than you need to be safe.
You have the case when the table does not have enough rows anyway.